lib/core/config.js

```diff
@@ -368,10 +372,31 @@ Config.prototype.loadEmbarkConfigFile = function() {
 };
 
 Config.prototype.loadPipelineConfigFile = function() {
-  var assets = this.embarkConfig.app;
-  for(var targetFile in assets) {
-    this.assetFiles[targetFile] = this.loadFiles(assets[targetFile]);
+
+  const defaultPipelineConfig = {
+    typescript: false
+  };
+
+  const pipelineConfigPath = fs.dappPath(
+    require(fs.dappPath(EMBARK_CONFIG_FILE)).config || DEFAULT_CONFIG_PATH,
+    PIPELINE_CONFIG_FILE
+  );
+
+  let pipelineConfig = defaultPipelineConfig;
+
+  if (fs.existsSync(pipelineConfigPath)) {
+    delete require.cache[pipelineConfigPath];
+    pipelineConfig = utils.recursiveMerge(
+      utils.recursiveMerge(true, pipelineConfig),
+      require(pipelineConfigPath)
+    );
   }
+
+  this.pipelineConfig = pipelineConfig;
+
+  Object.keys(this.embarkConfig.app).forEach(targetFile => {
+    this.assetFiles[targetFile] = this.loadFiles(this.embarkConfig.app[targetFile]);
+  });
```

Comment:
Loading asset files has been part of `loadPipelineConfigFile()` before. I just modernized the code a little bit.

What confused me though, was that prior to this commit, `loadPipelineConfigFile()` never loaded a config file, but asset files. So I'm wondering if it makes sense to put that algorithm in another `loadAssetFiles()` functions, or something along those lines that makes more semantic sense. 

Otherwise, now that `loadPipelineConfigFile()` actually **does** load a pipeline config file, we might as well just leave this as is and keep an eye on the fact that it does a little bit more (loading asset files).

Your call.

Reply:
Either way is fine with me, but I would have a slight preference for `loadAssetFiles` being its own function.